src/scene/webscene/index.jsx

```diff
@@ -39,21 +40,29 @@ class Webscene extends Component {
     if (!this.props.view || !this.props.view.map) return;
     this.componentIsMounted = false;
     this.props.view.map.remove(this.state.groupLayer);
+    this.props.view.map.ground.layers.remove(this.state.groundLayer);
   }
 
   update(prevProps = {}) {
-    if (!this.props.view || !this.state.groupLayer) return;
+    if (!this.props.view) return;
 
     if (prevProps.visible !== this.props.visible) {
-      this.state.groupLayer.visible = this.props.visible;
+      if (this.state.groupLayer) {
+        this.state.groupLayer.visible = this.props.visible;
+      }
+      if (this.state.groundLayer) {
+        this.state.groundLayer.visible = this.props.visible;
+      }
     }
 
     if (prevProps.layerSettings !== this.props.layerSettings) {
       Object.keys(this.props.layerSettings).forEach(layerId => {
         const settings = this.props.layerSettings[layerId];
-        const layer = this.state.groupLayer.layers.items.find(
-          l => l.id === layerId,
-        );
+        const layer =
+          this.state.groupLayer.layers.items.find(l => l.id === layerId) ||
+          (this.state.groundLayer.id === layerId
+            ? this.state.groundLayer
+            : undefined);
```

Comment:
```suggestion
        const layer = [...this.state.groupLayer.layers.items, this.state.groundLayer].find(l => l.id === layerId);
```

or, alternatively

```
        const layer = this.state.groundLayer.id === layerId ? this.state.groundLayer : this.state.groupLayer.layers.items.find(l => l.id === layerId);
```
(however the latter might now work with Dojo arrays, not sure)

Reply:
Could you give a comment why you didn't choose the more readable, first option?

Reply:
Hmm... I guess because I personally prefer ternary operators. Now that I am looking at it again though, I prefer the array find because it is much clearer what it is actually doing :) let me change that.